mesonbuild/modules/pkgconfig.py

```diff
@@ -338,18 +360,17 @@ def _generate_pkgconfig_file(self, state, deps, subdirs, name, description,
         else:
             outdir = state.environment.scratch_dir
             prefix = PurePath(coredata.get_option(mesonlib.OptionKey('prefix')))
-        # These always return paths relative to prefix
-        libdir = PurePath(coredata.get_option(mesonlib.OptionKey('libdir')))
-        incdir = PurePath(coredata.get_option(mesonlib.OptionKey('includedir')))
         fname = os.path.join(outdir, pcfile)
         with open(fname, 'w', encoding='utf-8') as ofile:
-            if not dataonly:
-                ofile.write('prefix={}\n'.format(self._escape(prefix)))
-                if uninstalled:
-                    ofile.write('srcdir={}\n'.format(self._escape(srcdir)))
-                if deps.pub_libs or deps.priv_libs:
-                    ofile.write('libdir={}\n'.format(self._escape('${prefix}' / libdir)))
-                ofile.write('includedir={}\n'.format(self._escape('${prefix}' / incdir)))
+            for optname in optnames:
+                if optname in referenced_vars:
```

Comment:
can't this be simplified to `for optname in referenced_vars` ?

Reply:
I think that's equivalent only if we add `prefix` in `referenced_vars` as suggested below.

Reply:
No, because sets are not ordered and I want to do this in the order that BUILTIN_DIR_OPTIONS returns. For the sake of neatness, and also because it is either that or `sorted(referenced_vars)` to ensure that the output is stable.

Reply:
mesonlib.OrderedSet?

Reply:
TBH that seems a solution in search of a problem.

Reply:
Actually mesonlib.OrderedSet would probably result in ugly looking files. It would be stable, but depending on the generate() invocation always have libdir/includedir out of order compared to `coredata.BUILTIN_DIR_OPTIONS`.

Reply:
ok, my comment was just a nitpicking to be honest. Since there is a reason for doing this that way, I'm fine with it.